packages/manager/modules/pci/src/projects/project/storages/cold-archive/add/components/steps/link-user-archive/controller.js

```diff
@@ -55,18 +54,35 @@ export default class ColdArchiveLinkUserArchiveController {
   }
 
   initUsersCredentialsList() {
-    this.usersCredentials = this.users.map((user) => ({
-      ...user,
-      credentialTrad: this.getCredentialTranslation(user),
-      userNameDescriptionKey: user.description
-        ? `${user.username} - ${user.description}`
-        : user.username,
-    }));
+    this.userModel.linkedMode.isInProgress = true;
+    this.pciStoragesColdArchiveService
+      .getAllS3Users(this.projectId)
+      .then((users) =>
+        this.pciStoragesColdArchiveService.mapUsersToCredentials(
+          this.projectId,
+          users,
+        ),
+      )
+      .then((usersWithCredentials) => {
+        this.usersCredentials = usersWithCredentials.map((user) => ({
+          ...user,
+          s3Credentials: user?.s3Credentials[0],
+          credentialTrad: this.getCredentialTranslation(user),
+          userNameDescriptionKey: user.description
+            ? `${user.username} - ${user.description}`
+            : user.username,
+        }));
+        this.users = this.usersCredentials;
```

Comment:
**question:** Why do you need same data on 2 different keys on the controller object? `this.users` and `this.usersCredentials`
```suggestion
        this.users = usersWithCredentials.map((user) => ({
          ...user,
          s3Credentials: user?.s3Credentials[0],
          credentialTrad: this.getCredentialTranslation(user),
          userNameDescriptionKey: user.description
            ? `${user.username} - ${user.description}`
            : user.username,
        }));
```